### DISCUSSION TITLE
DataTable with Entities that are subclasses

### DISCUSSION THREAD

**Discussion by ssalmeron** [primefaces/community#2373] [PrimeFaces] [answered]
Is it possible to display Entities that are subclasses of another?

Example: 

```java
@Entity
@Inheritance(strategy = InheritanceType.JOINED)
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "comments")
    private String comments;
}


public class PersonalAccount extends Account {

    @Column(name = "firstname")
    private String firstname;
}

public class CompanyAccount extends Account {    

    @Column(name = "companyName")
    private String companyname;
}
```

Above will create a Table for the Account with the shared fields, a table for PersonalAccount, and a table for CompanyAccount

em.createQuery("select A from Account A"); will return all Personal and Company accounts (using a join). 

Is it possible to display this structure in a Data Table? Considering that the columns will differ.

What other primeface UI Component can I use for this purpose?

**Comment by melloware** [comment, marked as answer]
Why not just flatten your selected object with a Projection or a wrapper pojo after selecting the results before sending to the DataTable so you can make both have `getName()`.   Or just overload each object to have a `getName()` that just points to the firstname or company name but in the DataTable you can just use `name`?

Lots of options.
